CMakeLists.txt

```diff
@@ -41,8 +41,32 @@ project(Csound)
 
 enable_testing()
 
-set(libcsound_private_include_dirs "")
-set(libcsound_public_include_dirs "")
+# only libcsound (for __BUILDING_LIBCSOUND)
+add_library(libcsound_only INTERFACE)
+target_compile_definitions(libcsound_only INTERFACE __BUILDING_LIBCSOUND)
+
+# for libcsound and plugins, utils, and tests that use internals
+add_library(libcsound_internals INTERFACE)
+target_include_directories(libcsound_internals INTERFACE
+    Engine H
+    ${CMAKE_CURRENT_BINARY_DIR} # for generated private headers
+)
+
+# for users
+add_library(libcsound_interface INTERFACE)
+target_include_directories(libcsound_interface INTERFACE
+    $<BUILD_INTERFACE:${Csound_SOURCE_DIR}/include>
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include> # for generated public headers
+    $<INSTALL_INTERFACE:include>
+)
+
+# for internal opcodes and plugins that use OpcodeBase.hpp
+add_library(opcode_internals INTERFACE)
+target_include_directories(opcode_internals INTERFACE Opcodes)
+
+# for utils that use std_util.h and utilmain.h
+add_library(utility_internals INTERFACE)
+target_include_directories(opcode_internals INTERFACE util)
```

Comment:
Is this meant to be `utility_internals ` instead of `opcode_internals`?

Reply:
Oops yeah fixed thanks!